engine/access/rpc/backend/backend_stream_transactions_test.go

```diff
@@ -0,0 +1,364 @@
+func (s *TransactionStatusSuite) TestSubscribeTransactionStatus() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	// Create subscription context with closer
+
+	// Generate sent transaction with ref block of the current finalized block
+	transaction := unittest.TransactionFixture()
+	transaction.SetReferenceBlockID(s.finalizedBlock.ID())
+	col := flow.CollectionFromTransactions([]*flow.Transaction{&transaction})
+	guarantee := col.Guarantee()
+	light := col.Light()
+	txId := transaction.ID()
+
+	expectedMsgIndexCounter := counters.NewMonotonousCounter(0)
+
+	// Create a special common function to read subscription messages from the channel and check converting it to transaction info
+	// and check results for correctness
+	checkNewSubscriptionMessage := func(sub subscription.Subscription, expectedTxStatus flow.TransactionStatus) {
+		unittest.RequireReturnsBefore(s.T(), func() {
+			v, ok := <-sub.Channel()
+			require.True(s.T(), ok,
+				"channel closed while waiting for transaction info:\n\t- txID %x\n\t- blockID: %x \n\t- err: %v",
+				txId, s.finalizedBlock.ID(), sub.Err())
+
+			txInfo, ok := v.(*convert.TransactionSubscribeInfo)
+			require.True(s.T(), ok, "unexpected response type: %T", v)
+
+			assert.Equal(s.T(), txId, txInfo.ID)
+			assert.Equal(s.T(), expectedTxStatus, txInfo.Status)
+
+			expectedMsgIndex := expectedMsgIndexCounter.Value()
+			assert.Equal(s.T(), expectedMsgIndex, txInfo.MessageIndex)
+			wasSet := expectedMsgIndexCounter.Set(expectedMsgIndex + 1)
+			require.True(s.T(), wasSet)
+		}, 50*time.Second, fmt.Sprintf("timed out waiting for transaction info:\n\t- txID: %x\n\t- blockID: %x", txId, s.finalizedBlock.ID()))
+	}
+
+	// 1. Subscribe to transaction status and receive the first message with pending status
+	subCtx, subCancel := context.WithCancel(ctx)
+	sub := s.backend.SubscribeTransactionStatuses(subCtx, &transaction.TransactionBody)
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusPending)
+
+	// 2. Make transaction reference block sealed, and add a new finalized block that includes the transaction
+	s.sealedBlock = s.finalizedBlock
+	s.addNewFinalizedBlock(s.sealedBlock.Header, func(block *flow.Block) {
+		block.SetPayload(unittest.PayloadFixture(unittest.WithGuarantees(&guarantee)))
+		s.collections.On("LightByID", mock.AnythingOfType("flow.Identifier")).Return(&light, nil).Maybe()
+	})
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusFinalized)
+
+	// 3. Add one more finalized block on top of the transaction block and add execution results to storage
+	finalizedResult := unittest.ExecutionResultFixture(unittest.WithBlock(s.finalizedBlock))
+	s.resultsMap[s.finalizedBlock.ID()] = finalizedResult
+
+	s.addNewFinalizedBlock(s.finalizedBlock.Header)
+	checkNewSubscriptionMessage(sub, flow.TransactionStatusExecuted)
+
+	// 4. Make the transaction block sealed, and add a new finalized block
+	s.sealedBlock = s.finalizedBlock
```

Comment:
`sealedBlock` should already equal `finalizedBlock` from step 2. is one of these incorrect?